### PR TITLE
Move env-var read logic to common dir, and optional reading of env-var

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

1. Move env var reading logic to sdk::common namespace, this allows it to be reused at multiple places.
2. Optional reading of env variables based on macro -DNO_GETENV. The env var is not read if the macro is set. This allows using sdk with platforms like UWP which doesn't support env variables. ( as discussed in #656 ).
3. Make required otlp-grpc exporter options to be read through env variable, which further allows loading this exporter at run-time using the dynamic-load plugin. This would be further added for other exporters also.

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

File: exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_exporter.h [hunk 1]
```diff
@@ -11,26 +11,56 @@
...
+
+  auto endpoint = opentelemetry::sdk::common::GetEnvironmentVariable(kOtlpGrpcEndpointEnv);
+  return endpoint.size() ? endpoint : kOtlpGrpcEndpointDefault;
+}
+
+inline const bool GetOtlpGrpcDefaultIsSslEnable()
+{
+  constexpr char kOtlpGrpcIsSslEnableEnv[] = "OTEL_EXPORTER_OTLP_GRPC_SSL_ENABLE";
+  auto ssl_enable = opentelemetry::sdk::common::GetEnvironmentVariable(kOtlpGrpcIsSslEnableEnv);
+  if (ssl_enable == "True" || ssl_enable == "TRUE" || ssl_enable == "true" || ssl_enable == "1")
```
Comment (ThomsonTan) [inline exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_exporter.h:35]: Convert it to lower case before comparison, to make the comparison case insensitive?

**Review by ThomsonTan — APPROVED** [review]

File: exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_exporter.h (hunk 1 above)
Comment (lalitb) [reply exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_exporter.h:35]: Good point will change that.

Comment (lalitb) [reply exporters/otlp/include/opentelemetry/exporters/otlp/otlp_grpc_exporter.h:35]: Ok. I realize there is no method available in `stdlib` to perform this conversion. we have to iterate over the string to convert it.  The existing comparison looks simple as compared to that :)

**Pull request merged**